### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'on')

### ISSUE THREAD

**Issue by gaurav-thakur-aurea** [rogalmic/vscode-bash-debug#170]
I am getting error while debugging bash script  with VS Code
Given Below is the Erroor
[2022-08-02 14:05:19.269] [renderer1] [error] TypeError: Cannot read properties of undefined (reading 'on')
    at BashDebugSession.disconnectRequest (c:\Users\Admin\.vscode\extensions\rogalmic.bash-debug-0.3.9\out\bashDebug.js:38:27)
    at BashDebugSession.dispatchRequest (c:\Users\Admin\.vscode\extensions\rogalmic.bash-debug-0.3.9\node_modules\vscode-debugadapter\lib\debugSession.js:385:22)
    at BashDebugSession.dispatchRequest (c:\Users\Admin\.vscode\extensions\rogalmic.bash-debug-0.3.9\node_modules\vscode-debugadapter\lib\loggingDebugSession.js:54:15)
    at BashDebugSession.handleMessage (c:\Users\Admin\.vscode\extensions\rogalmic.bash-debug-0.3.9\node_modules\vscode-debugadapter\lib\protocol.js:60:18)
    at BashDebugSession._handleData (c:\Users\Admin\.vscode\extensions\rogalmic.bash-debug-0.3.9\node_modules\vscode-debugadapter\lib\protocol.js:154:34)
    at Socket.<anonymous> (c:\Users\Admin\.vscode\extensions\rogalmic.bash-debug-0.3.9\node_modules\vscode-debugadapter\lib\protocol.js:79:44)
    at Socket.emit (node:events:390:28)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at Socket.Readable.push (node:internal/streams/readable:228:10): Error: TypeError: Cannot read properties of undefined (reading 'on')
    at BashDebugSession.disconnectRequest (c:\Users\Admin\.vscode\extensions\rogalmic.bash-debug-0.3.9\out\bashDebug.js:38:27)
    at BashDebugSession.dispatchRequest (c:\Users\Admin\.vscode\extensions\rogalmic.bash-debug-0.3.9\node_modules\vscode-debugadapter\lib\debugSession.js:385:22)
    at BashDebugSession.dispatchRequest (c:\Users\Admin\.vscode\extensions\rogalmic.bash-debug-0.3.9\node_modules\vscode-debugadapter\lib\loggingDebugSession.js:54:15)
    at BashDebugSession.handleMessage (c:\Users\Admin\.vscode\extensions\rogalmic.bash-debug-0.3.9\node_modules\vscode-debugadapter\lib\protocol.js:60:18)
    at BashDebugSession._handleData (c:\Users\Admin\.vscode\extensions\rogalmic.bash-debug-0.3.9\node_modules\vscode-debugadapter\lib\protocol.js:154:34)
    at Socket.<anonymous> (c:\Users\Admin\.vscode\extensions\rogalmic.bash-debug-0.3.9\node_modules\vscode-debugadapter\lib\protocol.js:79:44)
    at Socket.emit (node:events:390:28)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at o.handleErrorResponse (vscode-file://vscode-app/c:/Users/Admin/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/workbench.desktop.main.js:1047:45750)
    at vscode-file://vscode-app/c:/Users/Admin/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/workbench.desktop.main.js:1047:45189

I opend the bashDebug.js file on editor and see that its not able to find global variable **this.proxyProcess**.

Could you please help me to fix that issue 

Software Installed on My Machine
1) Ubantu 22.0.4 LTS
2) Vs Code
3)  Node.js v16.16.0

**Comment by gaurav-thakur-aurea** [comment]
![image](https://user-images.githubusercontent.com/22911897/182332982-d534d8e3-9b4d-42ca-aaf1-70f17c7076d0.png)

Failed here in bashdebug.js file
![image](https://user-images.githubusercontent.com/22911897/182333083-78d2109c-c7b0-4cb0-9d2c-9bbd2b44be16.png)

**Comment by gaurav-thakur-aurea** [comment]
![image](https://user-images.githubusercontent.com/22911897/182334314-c694d7c1-0a14-441e-bd28-bc2d13590c51.png)

**Comment by gaurav-thakur-aurea** [comment]
If i Run file with bash command it works fine inside of vs code

Please see Output

![image](https://user-images.githubusercontent.com/22911897/182334528-0ff1565b-2496-4a00-8ac9-c1ecd48322e4.png)

**Comment by gaurav-thakur-aurea** [comment]
I found the solution 
https://stackoverflow.com/questions/44049070/bash-on-ubuntu-on-windows-not-starting

I had the same issue but my solution was different. I found that my (Default) WSL distribution was set to docker-desktop-data instead of my Ubuntu distribution, e.g.

In PowerShell (Windows Shell): wslconfig /L

This lists WSL distributions. For me "docker-desktop-data (Default)" was listed as default, not my Ubuntu distribution.

To change this:

In PowerShell (Windows Shell): wslconfig /setdefault Ubuntu-20.04
